assets/js/googlesitekit/modules/datastore/modules.test.js

```diff
@@ -287,6 +288,31 @@ describe( 'core/modules modules', () => {
 			} );
 		} );
 
+		describe( 'registerModule', () => {
+			const moduleSlug = 'test-module';
+			const moduleSettings = {
+				name: 'Test Module',
+				order: 1,
+				description: 'A module for testing',
+				homepage: 'https://sitekit.withgoogle.com/',
+				icon: 'icon-name',
+			};
+
+			it( 'registers a module', async () => {
```

Comment:
It would be good to add a test (or add to this test) confirming anything stored outside the redux store (in the `ModuleComponents` variable) is stored here as well.